### ISSUE TITLE
DOC: `log10` documentation uses [-pi, pi] instead of (-pi, pi]

### ISSUE THREAD

**Issue by bersbersbers** [numpy/numpy#21235]
### Issue with current documentation:

> Logarithm is a multivalued function: for each x there is an infinite number of z such that 10**z = x. The convention is to return the z whose imaginary part lies in [-pi, pi].
https://numpy.org/doc/stable/reference/generated/numpy.log10.html

I wonder how `log10` can ever return both imaginary parts of pi and -pi. Since `np.imag(np.log(-1+0j)) == np.pi`, I'd say the correct interval should be (-pi, pi].

**Comment by seberg** [comment]
Exact intervals like this are often an illusion due to numerical accuracies.  Note that:
```
np.imag(np.log(-1 - 0.000000000000000000000001j)) == -np.pi
```
does return `True`.  I suspect that is the reason for how the documentation is written.  So my guess there is nothing much to do (or does it need clarification)?

**Comment by charris** [comment]
The inverse functions do have standardized ranges,  maybe we should use that with a note that numerical errors may lead to violations? Or maybe leave it as is with an explanation?

**Comment by bersbersbers** [comment]
> ```
> np.imag(np.log(-1 - 0.000000000000000000000001j)) == -np.pi
> ```

Thanks, this convinces me. I prefer an interval that I can use for range checking (even if it looks wrong theoretically) over a theoretical one that I'd need to code exceptions for to accommodate numerical inaccuracies.

**Comment by seberg** [comment]
Yeah, but I tend to agree with @charris.  It is probably clearer to write `(-pi, pi]` and then note that numerical accuracies mean that it is possible for the result to be so close to `-pi` that it evaluates as equal.

**Comment by bersbersbers** [comment]
@seberg yes, true. Especially since the statement refers to "the convention", which certainly is independent of numerical inaccuracies.

**Comment by seberg** [comment]
This was addressed by the PR above (but not closed automatically), closing.
